### PR TITLE
Bugfix/redirect url construct

### PR DESCRIPTION
**Pull request by icecream78**
According to issue #90  I've implemented logic about prioritization of redirect parameter into workflow of redirect urls construction.
Now, when function getRefferer() calls - firstly checked redirect_to header/parameter, if it's empty - check referrer header, if it's empty too - return config_site value. I suppose that logic fits customers wishes.
Additionally add new configuration parameter REDIRECT_PARAM_NAME for constomization alias of redirect url into input parameters list (default is redirect_to).

### CODE REVIEW

**Comment by icecream78** [comment]
@awalias  need your opinion and review

File: api/verify_test.go [hunk 1]
```diff
@@ -235,3 +235,72 @@ func (ts *VerifyTestSuite) TestVerifyNotPermitedCustomUri() {
...
+			uriAllowList:        []string{"http://localhost:3000"},
+			requestRedirectURL:  "http://localhost:3000/",
+			expectedRedirectURL: "http://localhost:3000/",
+		},
+		{
+			desc:                "different site url and not permited redirect url",
+			siteURL:             "https://someapp-something.codemagic.app/#/",
+			uriAllowList:        []string{},
+			requestRedirectURL:  "http://localhost:3000/#/",
+			expectedRedirectURL: "https://someapp-something.codemagic.app/",
```
Comment (awalias) [inline api/verify_test.go:273]: test cases can be:

```
		{
			desc:                "same site url and redirect url with path",
			siteURL:             "http://localhost:3000",
			uriAllowList:        []string{},
			requestRedirectURL:  "http://localhost:3000",
			expectedRedirectURL: "http://localhost:3000",
		},
		{
			desc:                "different site url and redirect url with path",
			siteURL:             "https://someapp-something.codemagic.app",
			uriAllowList:        []string{"http://localhost:3000"},
			requestRedirectURL:  "http://localhost:3000",
			expectedRedirectURL: "http://localhost:3000",
		},
		{
			desc:                "different site url and redirect url withput path",
			siteURL:             "https://someapp-something.codemagic.app",
			uriAllowList:        []string{"http://localhost:3000"},
			requestRedirectURL:  "http://localhost:3000/some/thing",
			expectedRedirectURL: "https://someapp-something.codemagic.app",
		},
		{
			desc:                "different site url and not permited redirect url",
			siteURL:             "https://someapp-something.codemagic.app",
			uriAllowList:        []string{},
			requestRedirectURL:  "https://someapp-something.codemagic.app/some/thing",
			expectedRedirectURL: "https://someapp-something.codemagic.app/some/thing",
		},
```

File: conf/configuration.go [hunk 2]
```diff
@@ -109,8 +109,9 @@ type MailerConfiguration struct {
 
 // Configuration holds all the per-instance configuration.
 type Configuration struct {
-	SiteURL       string                `json:"site_url" split_words:"true" required:"true"`
-	URIAllowList  []string              `json:"uri_allow_list" split_words:"true"`
+	SiteURL           string                `json:"site_url" split_words:"true" required:"true"`
+	URIAllowList      []string              `json:"uri_allow_list" split_words:"true"`
+	RedirectParamName string                `json:"redirect_param_name" default:"redirect_to"`
```
Comment (awalias) [inline conf/configuration.go:114]: I don't think we should include custom redirect param name in this PR

**Review by awalias — CHANGES_REQUESTED** [review]
I think we should fix the prioritization only, and in other PRs can address the other issues 

but looks good otherwise 👍

File: api/verify.go [hunk 3]
```diff
@@ -113,17 +114,19 @@ func (a *API) Verify(w http.ResponseWriter, r *http.Request) error {
...
 		if token != nil {
 			q := url.Values{}
 			q.Set("access_token", token.Token)
 			q.Set("token_type", token.TokenType)
 			q.Set("expires_in", strconv.Itoa(token.ExpiresIn))
 			q.Set("refresh_token", token.RefreshToken)
 			q.Set("type", params.Type)
-			rurl += "#" + q.Encode()
+
+			if strings.HasSuffix(rurl, "#") || strings.HasSuffix(rurl, "?") || strings.HasSuffix(rurl, "/") {
```
[the remark this answers is not in the document]
Comment (icecream78) [reply api/verify.go:125]: I guess that it's better add this check because if url ends with / then hardcoded # char will be encoded as %23

**Comment by icecream78** [comment]
I've add fixes according to review

File: api/verify.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (awalias) [reply api/verify.go:125]: let's fix this one in a different branch, keep this PR just for redirect fix

**Pull request merged**